### ISSUE TITLE
cloneModel is not being used for droping

### ISSUE THREAD

**Issue by pancho111203** [angular-ui-tree/angular-ui-tree#679]
I am using the data-clone-enabled attribute to clone elements from a Toolbox into a tree. The cloning works fine, but now I want to modify the element before it's dropped into the tree, as it needs different attributes. 

For this I want to modify the element that is going to go into the tree(dest) without modifying the element that is on the toolbox(source).

According to the documentation, the 'cloneModel' attribute is supposed to be the one attached into the destination:

>cloneModel: Given data-clone-enabled is true, holds the model of the cloned node that is to be inserted, this can be edited before drop without affecting the source node.

```javascript
 $scope.treeOptionsToolbox = {
                beforeDrop: function(e) {
                    console.log(e);

                    e.source.cloneModel.type = "sdfasdf";

                }
}
```

This doesn't work, as the target element keeps it's own value and cloneModel is ignored.

I have tried with this alternative:

```javascript
$scope.treeOptionsToolbox = {
                beforeDrop: function(e) {
                    console.log(e);

                    e.source.nodeScope.$modelValue.name = "sdfasdf";

                }
};
```

Now both the source and destination are modified.

**Comment by pancho111203** [comment]
Used 'dropped' instead of 'beforeDrop' and it works fine!

**Comment by pancho111203** [comment]
I ended up doing it like this:

```javascript
$scope.treeOptionsToolbox = {
                dropped: function(e) {
                    
                    // Convert the ToolboxElement into a TreeElement
                    var index = e.dest.index;
                    var modelDest: ToolboxElement = e.dest.nodesScope.$modelValue[index];

                    e.dest.nodesScope.$modelValue[index] = modelDest.insertIntoTree(TreeElement.getNewId());

                }

};
```

I'm not sure if cloneModel is working as intended though, therefore I will leave this open.

**Comment by Voles** [comment]
This issue has been fixed in the v2.12.0 release (see #627). @pancho111203 can you confirm updating works for you?

**Comment by pancho111203** [comment]
I updated and now it works, thank you!
